dashboard/app/models/async_progress_handler.rb

```diff
@@ -0,0 +1,30 @@
+# An SQS messages handler for asynchronous user level progress operations. These
+# include activities and user_level updates.
+
+class AsyncProgressHandler
+
+  # Returns a thread-local SQS queue for handle asynchronous user progress operations
+  # (Activity and UserLevel updates.) The queue is hread-local because the SQS client
+  # is not thread-safe.
+  def AsyncProgressHandler.progress_queue
+    Thread.current['progress_queue'] ||=
+        SQS::SQSQueue.new(Aws::SQS::Client.new, CDO.activity_queue_url)
+  end
+
+  def handle(messages)
+    ActiveRecord::Base.transaction do
+      messages.each do |message|
+        op = JSON.parse(message.body)
```

Comment:
These are the only two cases of async progress ops that I know of.  I like the explicitness of being able to read the code and know that the two kinds of async operations are activity and user.